src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets

```diff
@@ -463,6 +476,22 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <Target Name="_CollectAssetsToCompress" DependsOnTargets="ProcessPublishFilesForBlazor;ComputeBlazorExtensions">
+    <ItemGroup>
+      <AssetToCompress
+        Include="@(StaticWebAsset)"
+        Condition="'%(AssetKind)' != 'Build' and ('%(StaticWebAsset.AssetTraitName)' == 'BlazorWebAssemblyResource' or '%(StaticWebAsset.AssetTraitName)' == 'Culture')">
+        <ConfigurationName>BlazorPublishGzip</ConfigurationName>
+      </AssetToCompress>
+
+      <AssetToCompress
+        Include="@(StaticWebAsset)"
+        Condition="'%(AssetKind)' != 'Build' and ('%(StaticWebAsset.AssetTraitName)' == 'BlazorWebAssemblyResource' or '%(StaticWebAsset.AssetTraitName)' == 'Culture')">
+        <ConfigurationName>BlazorPublishBrotli</ConfigurationName>
+      </AssetToCompress>
+    </ItemGroup>
```

Comment:
Maybe we could combine these into one item, something like:
```targets
<AssetToCompress
  Include="@(StaticWebAsset)"
  Condition="'%(AssetKind)' != 'Build' and ('%(StaticWebAsset.AssetTraitName)' == 'BlazorWebAssemblyResource' or '%(StaticWebAsset.AssetTraitName)' == 'Culture')">
  <ConfigurationName>BlazorPublishGzip;BlazorPublishBrotli</ConfigurationName>
</AssetToCompress>
```

Reply:
Is there a reason why the patterns don't take care of these?

Reply:
Kind of. In this case, the criterion for compressing a file is expressed in terms of its `AssetTraitName` instead of its file name. We could maybe define a very specific pattern that includes the right file names and file extensions to perfectly match this subset (which I did try), but even this won't work if the customer defines their own `BlazorPublishExtension` item that doesn't conform to that pattern.